modules/ensemble/lib/widget/input/slider.dart

```diff
@@ -124,27 +128,56 @@ class SliderState extends FormFieldWidgetState<EnsembleSlider> {
-              divisions: widget.controller.divisions,
-              onChanged: isEnabled()
-                  ? (value) {
-                      setState(() {
-                        widget.controller.value = value;
-                      });
-                      if (widget.controller.onChange != null) {
-                        ScreenController().executeAction(
-                          context,
-                          widget.controller.onChange!,
-                          event: EnsembleEvent(widget),
-                        );
-                      }
-                    }
-                  : null,
-            ),
+            child: widget.controller.isRange
+                ? RangeSlider(
+                    labels: RangeLabels(
+                      widget.controller.value.start
+                          .toStringAsFixed(decimalPlaces),
+                      widget.controller.value.end
+                          .toStringAsFixed(decimalPlaces),
+                    ),
+                    min: widget.controller.minValue,
+                    max: widget.controller.maxValue,
+                    values: widget.controller.value,
+                    divisions: widget.controller.divisions,
+                    onChanged: isEnabled()
+                        ? (value) {
+                            setState(() {
+                              widget.controller.value = value;
+                            });
+                            if (widget.controller.onChange != null) {
+                              ScreenController().executeAction(
+                                context,
+                                widget.controller.onChange!,
+                                event: EnsembleEvent(widget),
+                              );
+                            }
+                          }
+                        : null,
+                  )
+                : Slider(
+                    label: widget.controller.value.start
+                        .toStringAsFixed(decimalPlaces),
+                    min: widget.controller.minValue,
+                    max: widget.controller.maxValue,
+                    value: widget.controller.value.start,
+                    divisions: widget.controller.divisions,
+                    onChanged: isEnabled()
+                        ? (value) {
+                            setState(() {
+                              widget.controller.value =
+                                  RangeValues(value, value);
+                            });
+                            if (widget.controller.onChange != null) {
+                              ScreenController().executeAction(
+                                context,
+                                widget.controller.onChange!,
+                                event: EnsembleEvent(widget),
+                              );
+                            }
+                          }
+                        : null,
+                  ),
```

Comment:
Please remove the duplicate code which can be converted to method and use them.

Reply:
Move whole onChanged function to a separate function